include/proxy-wasm/word.h

```diff
@@ -24,11 +24,12 @@ namespace proxy_wasm {
 // Use byteswap functions only when compiling for big-endian platforms.
 #if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) &&                                    \
     __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
-#define htowasm(x) __builtin_bswap32(x)
-#define wasmtoh(x) __builtin_bswap32(x)
+#define htowasm1111(x, is_null) is_null ? (x) : __builtin_bswap32(x)
+#define htowasm(x, is_wasm_byte_order) is_wasm_byte_order ? __builtin_bswap32(x) : (x)
+#define wasmtoh(x, is_wasm_byte_order) is_wasm_byte_order ? __builtin_bswap32(x) : (x)
```

Comment:
You need to add surrounding brackets, otherwise it might interact with surrounding codebase in unexpected ways (it's a macro, not a function).

Reply:
Added surrounding brackets